bitnami/tomcat/templates/_pod.tpl

```diff
@@ -77,13 +77,21 @@ containers:
     env:
       - name: BITNAMI_DEBUG
         value: {{ ternary "true" "false" .Values.image.debug | quote }}
+      {{- if and .Values.existingSecret .Values.secretKeys.adminUsernameKey }}
+      - name: TOMCAT_USERNAME
+        valueFrom:
+          secretKeyRef:
+            name: {{ include "tomcat.secretName" . }}
+            key: {{ include "tomcat.adminUsernameKey" . }}
+      {{- else }}
       - name: TOMCAT_USERNAME
         value: {{ .Values.tomcatUsername | quote }}
+      {{- end }}
```

Comment:
Even better. All this logic could be implemented in the _helper function. Note that it checks `.Values.secretKeys.adminUsernameKey` twice.